### ISSUE TITLE
[Visual Studio] New Function App does not respect .Net version

### ISSUE THREAD

**Issue by justcla** [Azure/Azure-Functions#489]
User (Ben Edwards) raised issue on Marketplace:

> Create a new FunctionApp project with default settings, targeting 4.7.
> * Creating a new function app *still* does not respect you setting for .net version (ask for 4.7, sets project to 4.6.1, but sets build flags to 4.7)

**Comment by justcla** [comment]
This is a known issue. And it's not specific to Azure Functions alone. It's an issue with the New Project dialog. There are many templates for which the .net version drop-down does not apply (ie. C++ Projects).

Azure Functions currently only work with 4.6.1, so all Azure Functions projects will be set to 4.6.1 regardless of what was showing in the .Net version selector in the New Project dialog at the time the project was created.

So, you can ignore the .Net version selector in the new project dialog for now.
We are working on a solution to avoid this confusion.

Thanks for raising the issue.

**Comment by elexisvenator** [comment]
> Azure Functions currently only work with 4.6.1, so all Functions projects need to target v4.6.1 - regardless of what was showing in the .Net version selector in the New Project dialog at the time the project was created.

Is this true though? I can manually retarget the project to 4.7, us C#7.1 Features and publish and it works fine (havent tested with .Net 4.7 specific things like ECC enhancements). Also it is possible to target .Net 4.7 in your Function App on the Azure portal.

**Comment by paulbatum** [comment]
Yeah this is not true. You can target an Azure Function project to 4.6.1/4.6.2/4.7. It will then run on whatever version of the framework you have installed on the machine. In the case of an function app created in Azure, that will be 4.7 (as you mentioned, the portal selector shows this).

**Comment by hablutzel1** [comment]
@justcla, as of February 2018, Azure Functions seems to have updated to use .NET 4.7.1 by default.

Is it currently possible to choose a previous .NET framework version to be used locally during development and later after publishing the Function App to Azure?.

**Comment by justcla** [comment]
Not sure. Let me check with the team.

**Comment by maximusgdb** [comment]
@paulbatum  I don't find the " target framework dropdown in the project settings" of my Azure Function project in the Azure Portal. Could you help me on that? Thanks

**Comment by paulbatum** [comment]
@maximusgdb There is no UX for selecting this in the portal. I was referring to the setting in Visual Studio.

**Comment by maximusgdb** [comment]
@paulbatum  ok thanks. And so is there away to change the target framework in the Azure Portal? I don't find anything about it...

**Comment by azharuddinsayed** [comment]
i have one web application target to .net framework 4.5 and form that application i want to call a http trigger azure function it is not allowing me to call , it is giving me error **An existing connection was forcibly closed by the remote host.** Please help me .
